### ISSUE TITLE
`""` results in instance of outer `String`

### ISSUE THREAD

**Issue by ppibburr** [mattn/mruby-thread#36]
``` ruby
Thread.new do
  "".is_a?(String) # => false
end
```

**Comment by ppibburr** [comment]
Further more:
```ruby
  def foo; nil; end

Thread.new do
  "".methods.index("foo")                       #=> (>= 0)
  String.instance_methods.index("foo") #=> nil
end
```

**Comment by mattn** [comment]
strange. it should be fixed in https://github.com/mattn/mruby-thread/commit/e139cfed05bbcbd5a37ff46c9994933cdd926e3e

**Comment by ppibburr** [comment]
`''` and `%q` also affected

```
$ build/host/bin/mruby -e "Thread.new do p ''.is_a?(String) end.join"
false
$ build/host/bin/mruby -e "Thread.new do p %q[w].is_a?(String) end.join"

```

**Comment by mattn** [comment]
should be fixed by #36
